src/SkyApm.Core/Common/CpuHelpers.cs

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the SkyAPM under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The SkyAPM licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SkyApm.Common
+{
+    internal static class CpuHelpers
+    {
+        private const int interval = 1000;
+        private static double _usagePercent;
+        private static readonly Task _task;
+
+        public static double UsagePercent => _usagePercent;
+
+        static CpuHelpers()
+        {
+            var process = Process.GetCurrentProcess();
+            _task = Task.Factory.StartNew(async () =>
+            {
+                var _prevCpuTime = process.TotalProcessorTime.TotalMilliseconds;
+                while (true)
+                {
+                    var prevCpuTime = _prevCpuTime;
+                    var currentCpuTime = process.TotalProcessorTime;
+                    var usagePercent = (currentCpuTime.TotalMilliseconds - prevCpuTime) / interval;
+                    Interlocked.Exchange(ref _prevCpuTime, currentCpuTime.TotalMilliseconds);
+                    Interlocked.Exchange(ref _usagePercent, usagePercent);
+                    await Task.Delay(1000);
+                }
+            }, TaskCreationOptions.LongRunning);
```

Comment:
need TaskCreationOptions.LongRunning ?

https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/Threading/Tasks/ThreadPoolTaskScheduler.cs#L47

Reply:
This task refreshes CPU usage every second until the application exit, so it should use `LongRunning` mode.

Reply:
```
       protected internal override void QueueTask(Task task)
        {
            TaskCreationOptions options = task.Options;
            if ((options & TaskCreationOptions.LongRunning) != 0)
            {
                // Run LongRunning tasks on their own dedicated thread.
                Thread thread = new Thread(s_longRunningThreadWork);
                thread.IsBackground = true; // Keep this thread from blocking process shutdown
                thread.Start(task);
            }
            else
            {
                // Normal handling for non-LongRunning tasks.
                bool preferLocal = ((options & TaskCreationOptions.PreferFairness) == 0);
                ThreadPool.UnsafeQueueUserWorkItemInternal(task, preferLocal);
            }
        }
```
When an async method reaches an await for Task.Delay(1000), the thread (just created background thread, not ThreadPool thread) will be released. when the task is completed the rest logic will be scheduled in ThreadPool thread. this means that we created a new thread needlessly.